6.5.0

anoblet opened the pull request:
Summary:

* Merge 6.5.0
* Use upstream version of `useProtectedMedia.ts` and add customizations
* Combine `VideoDetailsInline.tsx`
* Temporarily remove customizations in `Layout.tsx` and `Header.tsx`
* Use upstream version of `Layout.tsx`
* Reimplement login and account buttons
* Add `APP_OAUTH_ENABLED` environment variable
* Disable the language switcher if OAuth is enabled

Commit:
chore: merge release to develop

ChristiaanScheermeijer committed:
feat: add ellipsis to card title

ChristiaanScheermeijer committed:
feat(menu): support more items in header navigation

ChristiaanScheermeijer committed:
fix(videodetail): buttons wrapping

mirovladimitrovski committed:
Remove the InPlayer SDK usage from the Web App and use API endpoints instead (#569)

ChristiaanScheermeijer committed:
feat(project): add cancel functions for debounce and throttle utils

mirovladimitrovski committed:
feat(integrations): replace InPlayer SDK with direct API calls to JwPlayer SIMS domain (#578)

langemike committed:
feat: add backClick event support for player

* feat: add backClick event support for player

* chore: useRef for backclick performed

* chore: restore backClickRef value & reintroduce detachEvents

---------

Co-authored-by: Mike van Veenhuijzen <mike@videodock.com>

mirovladimitrovski committed:
feat(profiles): remove all remaining dead code assotiated with profiles

ChristiaanScheermeijer committed:
feat: implement tile-slider dependency

langemike committed:
fix: wait for geo status and cache it's value

Co-authored-by: Mike van Veenhuijzen <mike@videodock.com>

AntonLantukh committed:
feat(project): add ssai ads for vod (#583)

* feat(project): add ssai ads for vod, refactor sources function

Commit:
chore(release): v6.5.0

AntonLantukh committed:
Merge branch 'release-candidate' into release

anoblet committed:
Merge tag 'v6.5.0' into anoblet/feat/6.5.0

anoblet commented on packages/ui-react/src/components/UserMenu/UserMenu-CXL.tsx:
```diff
@@ -0,0 +1,98 @@
+
+  const { login: oAuthLogin } = useOAuth();
+
+  const { config } = useConfigStore(({ config, accessModel, supportedLanguages }) => ({ config, accessModel, supportedLanguages }), shallow);
```
This is a lot of code to get `isOAuthMode` where an environment variable would work. I don't see a scenario where the app, and JWP would have a different setting. My suggestion is to add `APP_OAUTH_ENABLED` as a boolean to env and remove it from JWP(once refactoring has been finished).

Current:

```
const { config } = useConfigStore(({ config, accessModel, supportedLanguages }) => ({ config, accessModel, supportedLanguages }), shallow);

const { custom } = config;

const isOAuthMode = custom?.isOAuthMode;
```

Suggested:

```
const isOAuthMode = env.APP_OAUTH_ENABLED;
```

pawelkmpt replied:
This is the commit where it was introduced: https://github.com/conversionxl/ott-web-app/commit/d584a044d89b9624593b6170191c1bacd1df4d9d

From my perspective using `env` is fine. I'm not able to tell anything from the React or JWP perspective.
@anoblet test it and see if works well. If works let's use `env`

anoblet committed:
feat: reimplement custom login and back to account buttons

anoblet commented on packages/ui-react/src/containers/Layout/Layout-CXL.tsx:
```diff
@@ -0,0 +1,90 @@
+    })),
+  ];
+
+  if (config.custom?.navItemAccount) {
```
I can implement before, after, very quickly though it seems over complicated. How often does the text/url change, can we hard code it?

anoblet commented:
There's customizations in place for before, after, and right items to be added to the header. We seem to be using only one after item(external link to the signup page). Does it make to keep this customization in place? I've added the after logic, though since before and right aren't used I haven't added those back yet.

https://github.com/conversionxl/ott-web-app/blob/f468ee8b033a504b98fbf90531262ee22a2cee88/packages/ui-react/src/containers/Layout/Layout-CXL.tsx#L60

pawelkmpt reviewed (commented):
> There's customizations in place for before, after, and right items to be added to the header. We seem to be using only one after item(external link to the signup page). Does it make to keep this customization in place? I've added the after logic, though since before and right aren't used I haven't added those back yet.

It was developed to give possibility to add menu items in the JWP dashboard without a need for using developer. Please keep it.

anoblet commented:
I added `before`, `after`, and `right` menu items. We need to update the value in JWP since currently it is an object, and it needs to be an array of objects.

anoblet committed:
feat: reimplement custom navigation

anoblet commented:
I updated the navigation to use the current syntax.

pawelkmpt commented:
> * Temporarily remove customizations in `Layout.tsx` and `Header.tsx`
> * Use upstream version of `Layout.tsx`

I compared `Layout.tsx` and `Layout-CXL.tsx` and they look completely different.

We talked about significant upstream changes and agreed on:
* keeping `Layout.tsx` component same as upstream
* creating `Layout-CXL.tsx` based on the upstream component
* applying customizations on top of new layout component in `Layout-CXL.tsx`

I have an impression `Layout-CXL.tsx` is simply copy-pasted version of the old component not new layout with customizations on top. Correct me if I'm wrong.

<img width="1624" alt="Screenshot 2025-03-04 at 14 39 15" src="https://github.com/user-attachments/assets/d6884150-29fd-4296-8a02-4ab386f9da07" />

pawelkmpt reviewed (changes requested):
This appears for a while when I'm signed in. It should not be visible at all.

<img width="136" alt="Screenshot 2025-03-04 at 14 57 49" src="https://github.com/user-attachments/assets/c1b882b2-060a-467e-bd0d-9e87a86a3967" />
<img width="258" alt="Screenshot 2025-03-04 at 14 57 52" src="https://github.com/user-attachments/assets/147d6477-f314-417a-8f29-f5dfbb9f565a" />

**Buttons arrangement**
6.4.0
<img width="737" alt="Screenshot 2025-03-04 at 14 59 59" src="https://github.com/user-attachments/assets/527cda0d-7823-4ca0-84b1-ce3e302b4ceb" />

6.5.0
<img width="788" alt="Screenshot 2025-03-04 at 15 00 05" src="https://github.com/user-attachments/assets/0e3f3bac-a7d7-42cc-93f4-3ccfeb735363" />

I asked Hesh about the version he'd like to have.

anoblet commented:
> I compared Layout.tsx and Layout-CXL.tsx and they look completely different.

I'm not sure which file you are looking at? This is the link to the `Layout-CXL.tsx` file in this branch: https://github.com/conversionxl/ott-web-app/blob/anoblet/feat/6.5.0/packages/ui-react/src/containers/Layout/Layout-CXL.tsx

> This appears for a while when I'm signed in. It should not be visible at all.

I disabled the header action button when OAuth is enabled.

anoblet committed:
feat: disable language switcher if oauth is enabled

anoblet committed:
feat: disable header action button if oauth is enabled

pawelkmpt commented:
> > I compared Layout.tsx and Layout-CXL.tsx and they look completely different.
> 
> I'm not sure which file you are looking at? This is the link to the `Layout-CXL.tsx` file in this branch: https://github.com/conversionxl/ott-web-app/blob/anoblet/feat/6.5.0/packages/ui-react/src/containers/Layout/Layout-CXL.tsx

Hmm, maybe I had cached file opened in the editor or wrong active branch. All good 👍 

> > This appears for a while when I'm signed in. It should not be visible at all.
> 
> I disabled the header action button when OAuth is enabled.

👍

pawelkmpt reviewed (changes requested):
Desktop works good, mobile menu has these JWP items which you recently hid on  desktop.

<img width="443" alt="Screenshot 2025-03-05 at 17 11 43" src="https://github.com/user-attachments/assets/e24715c6-7e78-47c8-9f3d-bca23cfc6ae9" />

anoblet commented:
I removed the payment button, and updated the account button to point to CXL dashboard. I also updated the login button(when signed out) to use the OAuth mechanism.

pawelkmpt reviewed (changes requested):
I still see issues on mobile. Please make sure you tested it properly.

**Logged out state**

![Screenshot 2025-03-06 at 16 25 11](https://github.com/user-attachments/assets/350c5e95-7b94-41f6-8ded-293e919d0d89)

Missing: before/after navigation items (they're present on lite.cxl.com)

**Logged in state**

Same as above + unnecessary settings section

![Screenshot 2025-03-06 at 16 25 43](https://github.com/user-attachments/assets/c0e3421b-ad81-4b05-9496-d6dee8c516e7)

anoblet commented:
Sidebar should be fixed.

Logged in:

![localhost_8080_(iPhone SE) (3)](https://github.com/user-attachments/assets/468b886a-edfd-4365-83e8-55bc71ab2fd1)

Logged out:
![localhost_8080_(iPhone SE) (4)](https://github.com/user-attachments/assets/411b8609-866c-47b3-8171-d07e2374fecc)

anoblet committed:
feat: reimplement sidebar customizations

anoblet committed:
feat: normalize cta button size

pawelkmpt reviewed (changes requested):
I was happy it's ready to go live and deployed it. Then found out Sign Up button doesn't work properly 😞 

It should redirect users to CXL site

![Screenshot 2025-03-07 at 10 33 10](https://github.com/user-attachments/assets/0de010e1-a8af-420c-82ac-a3fa0d0d556f)

![Screenshot 2025-03-07 at 10 32 57](https://github.com/user-attachments/assets/1aacfaa2-91d0-450d-92c8-a24e42356094)

anoblet committed:
feat: reimplement signup logic

anoblet commented:
I missed moving `HeaderUserMenu` to `HeaderUserMenu-CXL`.

pawelkmpt reviewed (commented):
@anoblet it's live. I'll merge in the afternoon if everything goes well. Ping me before you start working on the new version.

But I noticed search doesn't work properly. I'm not sure at which point it broke. I'll share a video later to show you what exactly is broken
